test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/RequestTests.cs

```diff
@@ -184,6 +187,51 @@ public Task RemoteIPv6Address()
             }
         }
 
+        [Fact]
+        public async Task ConnectionResetAbortsRequest()
+        {
+            const int connectionErrorEventId = 14; // from KestrelTrace.cs
+
+            var testSink = new TestSink(write => write.EventId.Id == connectionErrorEventId);
+            var requestAbortedAcquired = new CancellationTokenSource();
+            var builder = new WebHostBuilder()
+                .UseLoggerFactory(new TestLoggerFactory(testSink, true))
+                .UseKestrel()
+                .UseUrls($"http://127.0.0.1:0")
+                .Configure(app => app.Run(context =>
+                {
+                    return Task.FromResult(0);
```

Comment:
We should also assert that `context.Request.Body.ReadAsync` fails with an `IOException`.

I think that `SocketInputExtensions.ReadAsync` and possibly a few other places will have to be updated to check if there is an error (possibly by calling `SocketInput.GetResult()`) in a thread-safe way before exiting due to `RemoteIntakeFin` being set.